extensions/ql-vscode/src/variant-analysis/run-remote-query.ts

```diff
@@ -58,84 +60,119 @@ interface GeneratedQueryPack {
       language,
       packRelativePath,
     );
-  }
-  if (!language) {
-    throw new UserCancellationException("Could not determine language.");
-  }
-
-  // Clear the cliServer cache so that the previous qlpack text is purged from the CLI.
-  await cliServer.clearCache();
+    // Clear the cliServer cache so that the previous qlpack text is purged from the CLI.
+    await cliServer.clearCache();
+
+    // Install packs, since we just synthesized a dependency on the language's standard library.
+    needsInstall = true;
+  } else if (!cliSupportsMrvaPackCreate) {
+    // We need to copy the query pack to a temporary directory and then fix it up to work with MRVA.
+    queryPackDir = tmpDir.queryPackDir;
+    await copyExistingQueryPack(
+      cliServer,
+      originalPackRoot,
+      queryFile,
+      queryPackDir,
+      packRelativePath,
+    );
 
-  let precompilationOpts: string[] = [];
-  if (await cliServer.cliConstraints.usesGlobalCompilationCache()) {
-    precompilationOpts = ["--qlx"];
+    // We should already have all the dependencies available, but these older versions of the CLI
+    // have a bug where they will not search `--additional-packs` during validation in `codeql pack bundle`.
+    // Installing the packs will ensure that any extension packs get put in the right place.
+    needsInstall = true;
   } else {
-    const ccache = join(originalPackRoot, ".cache");
-    precompilationOpts = [
-      "--qlx",
-      "--no-default-compilation-cache",
-      `--compilation-cache=${ccache}`,
-    ];
+    // The CLI supports creating a MRVA query pack directly from the source pack.
+    queryPackDir = originalPackRoot;
+    // We expect any dependencies to be available already.
+    needsInstall = false;
   }
 
-  if (await cliServer.useExtensionPacks()) {
-    await injectExtensionPacks(cliServer, queryPackDir, workspaceFolders);
-  }
+  if (needsInstall) {
+    // Install the dependencies of the synthesized query pack.
+    await cliServer.packInstall(queryPackDir, {
+      workspaceFolders,
+    });
 
-  await cliServer.packInstall(queryPackDir, {
-    workspaceFolders,
-  });
+    // Clear the CLI cache so that the most recent qlpack lock file is used.
+    await cliServer.clearCache();
+  }
 
   // Clear the CLI cache so that the most recent qlpack lock file is used.
   await cliServer.clearCache();
+  if (cliSupportsMrvaPackCreate) {
+    precompilationOpts = [
+      "--mrva",
+      "--query",
+      join(queryPackDir, packRelativePath),
+      // We need to specify the extension packs as dependencies so that they are included in the MRVA pack.
+      // The version range doesn't matter, since they'll always be found by source lookup.
+      ...extensionPacks.map((p) => `--extension-pack=${p}@*`),
+    ];
+  } else {
+    if (await cliServer.cliConstraints.usesGlobalCompilationCache()) {
+      precompilationOpts = ["--qlx"];
+    } else {
+      const ccache = join(originalPackRoot, ".cache");
```

Comment:
```suggestion
      const cache = join(originalPackRoot, ".cache");
```
Typo, or is this intentional?

Reply:
This is the existing naming, I guess because it stands for compilation cache?

Reply:
I suspect it was a typo. Fixed.